src/sentry/api/decorators.py

```diff
@@ -19,11 +19,7 @@ def is_considered_sudo(request: Request) -> bool:
         or is_api_key_auth(request.auth)
         or is_api_token_auth(request.auth)
         or is_org_auth_token_auth(request.auth)
-        or (
-            request.user.is_authenticated
-            and not isinstance(request.user, AnonymousUser)
-            and not request.user.has_usable_password()
-        )
+        or (request.user.is_authenticated and not request.user.has_usable_password())
```

Comment:
`is_authenticated` can now be used to descriminate the `request.user` union -- so mypy is mad about the impossibility of `AnonymousUser` after that check

Reply:
> is_authenticated can now be used to descriminate the request.user union

This will be very useful :clap: